src/Sylius/Bundle/UiBundle/DependencyInjection/Compiler/TwigComponentTagPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\UiBundle\DependencyInjection\Compiler;
+
+use InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+final class TwigComponentTagPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        foreach ($container->findTaggedServiceIds('sylius.twig_component') as $id => $tags) {
+            foreach ($tags as $tag) {
+                $liveComponentService = $container->getDefinition($id);
+                $liveComponentService->addTag('twig.component', [
+                    'key' => $tag['key'] ?? throw new InvalidArgumentException('The "key" attribute is required for the "sylius.twig_component" tag'),
+                    'template' => $tag['template'] ?? throw new InvalidArgumentException('The "template" attribute is required for the "sylius.twig_component" tag'),
```

Comment:
Maybe extract checking the required keys before and throw one exception containing them both?
As it is, if I don't configure anything I'll get an exception w/ `key` and only after setting it I'll get the next one regarding the `template`.